lddwrap/__init__.py

```diff
@@ -0,0 +1,222 @@
+        commands = ["ldd", "--unused", path.as_posix()]
+
+        proc_unused = subprocess.Popen(
+            commands, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True, env=env)
+
+        out_unused, err_unused = proc_unused.communicate()
+        # return code = 0 -> no unused dependencies, return code = 1 -> some unused dependencies
+        if proc_unused.returncode not in [0, 1]:
+            raise RuntimeError("Failed ldd-ing external libraries of {} with code {}:\nout:\n{}\n\nerr:\n{}".format(
+                path, proc.returncode, out_unused, err_unused))
+
+        dependencies = _update_unused(dependencies=dependencies, out_unused=out_unused)
+
+    return dependencies
+
+
+def _dependencies_parse_helper(out: str, verbose: Optional[bool] = None) -> List[Dependency]:
+    """
+    Help to parse the command line output.
+
+    :param out: command line output
+    :param verbose if set, print all the information
+    """
+    if verbose:
+        print(out)
+
+    dependencies = []  # type: List[Dependency]
+    for line in [line.strip() for line in out.split('\n') if line.strip()]:
+        # end of output
+        if "Version information" not in line:
+            dep = parse_line(line=line)
+            if dep is not None:
+                dependencies.append(dep)
+        else:
+            break
+
+    return dependencies
+
+
+def _update_unused(dependencies: List[Dependency], out_unused: str) -> List[Dependency]:
+    """
+    Set Dependency argument unused.
+
+    :param dependencies: List of dependencies
+    :param out_unused: output from command ldd --unused
+    :return: updated list of dependencies
+    """
+    unused_dependencies = []  # type: List[pathlib.Path]
+    for line in [line.strip() for line in out_unused.split('\n') if line.strip()]:
+        # skip first line because it's no dependency
+        if line != "Unused direct dependencies:":
+            unused_dependencies.append(pathlib.Path(line.strip()))
+
+    for dep in dependencies:
+        dep.unused = False
+        for unused_dep in unused_dependencies:
+            if unused_dep == dep.path:
+                dep.unused = True
+                break
+
+    return dependencies
+
+
+def version(env: Optional[Dict[str, str]] = None) -> List[int]:
+    """
+    Return version of ldd.
+
+    :param env: the environment to use
+    :return: version X.Y
+    """
+    proc = subprocess.Popen(
+        ["ldd", "--version"], stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True, env=env)
+
+    out, err = proc.communicate()
+    if proc.returncode != 0:
+        raise RuntimeError("Failed retrieving ldd version with code {}:\nout:\n{}\n\nerr:\n{}".format(
+            proc.returncode, out, err))
+
+    # version is last element of first line
+    splitted_lines = out.split('\n', 1)
+
+    if len(splitted_lines) > 0:
+        first_line = splitted_lines[0]
+    else:
+        raise ValueError("ldd --version had no output")
```

Comment:
Whenever you return an error you should show the user what was wrong. the best would be if you print the message like this:

```
raise ValueError("ldd --version had no output: {!R}".format(splitted_lines))
```

Please note the `{!r}`. The `!r` prints the raw string, so the user will be able to see if the output was actually empty `''`